src/gui/notifications/dbusnotifier.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2015 Eugene Shalygin
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "dbusnotifier.h"
+
+#include "notifications.h"
+#include "core/bittorrent/torrenthandle.h"
+
+#include <QDesktopServices>
+#include <QDir>
+#include <QUrl>
+#include <QWidget>
+#include <QDBusConnection>
+
+
+namespace
+{
+    const QLatin1String ACTION_NAME_DEFAULT("activate-context");
+    const QLatin1String ACTION_NAME_OPEN_FINISHED_TORRENT("document-open"); // it must be named as the corresponding FDO icon
+    const QLatin1String DBUS_SERVICE_NAME("org.freedesktop.Notifications");
+    const QLatin1String DBUS_INTERFACE_NAME("org.freedesktop.Notifications");
+    const QLatin1String DBUS_PATH("/org/freedesktop/Notifications");
+}
+
+DBusNotifier::DBusNotifier(QWidget *mainWindow, QObject *parent)
+    : Notifier(parent)
+    , mainWindow(mainWindow)
```

Comment:
Disagree. DBus notifications may be used without GUI  (from qbittorrent-nox). Let's make this next step easier.